crates/worktree/src/worktree.rs

```diff
@@ -3174,6 +3200,12 @@ pub struct Entry {
     /// exclude them from searches.
     pub is_ignored: bool,
 
+    /// Whether this entry is always included in searches.
+    ///
+    /// This is used for entries that are always included in searches, even
+    /// if they are ignored by git or by `file_scan_exclusions`.
+    pub is_always_included: bool,
```

Comment:
This isn't the approach we'd like to take for this feature. `Entry` should only be a representation of all of the local file system state we care about. Conceptually, `file_scan_inclusions` should be an entirely client-local concern, implemented as query on top of this state. We shouldn't need to involve collab or the database to implement this feature.

`file_scan_exlcusions` doesn't work this way because it's more of an optimization for common types of files. I think it's ok for `file_scan_exclusions` to trump `file_scan_inclusions`.

Reply:
Could you elaborate more on the approach of `client-local` concern? I'm not sure I totally understand. There needs to be some way to modify the indexing behavior of Zed to force entries to always be included in its internal representation of the filesystem.

Another way I could see doing this without needing to modify the database schema or send anything over the wire is just changing all of the current call sites from `entry.is_always_included` to something like `self.settings.is_path_always_included(path)`. Is that more along the lines of what you were thinking?

Thanks for taking a look and letting me know!

Reply:
What I mean by 'client local', is that this setting shouldn't need to be synced over the wire as an additional field on the `Entry`. Zed user Alice should read all the files as they do now (excluding files matching `file_scan_exclusions`), and then when rendering or otherwise querying this data structure, Alice should check this setting and whether it's git ignored. Meanwhile, Zed user Bob (who Alice is currently pairing with), should receive all of the `Entry`s, like we're doing now, and then when rendering or otherwise working with these files, it locally computes whether the file should be included by combining his local `file_scan_inclusions` with the git ignored state. 

So, having written that out, I think I am recommending the second approach you mentioned :)

Does that answer the question?

Reply:
Yes, that makes sense now. Thanks for clarifying! I'll go ahead and make those changes and let you know when this PR is ready for review again.

Reply:
Awesome :D just mark it as ready for review when you're ready.

Reply:
Alright, this should be good to go. I ended up having to either pass around the `WorktreeSettings` struct to some functions or add it as a field to some other structs, which I'm a bit reticent to do for some reason, but the database/collab should no longer be involved. Let me know if there's any other concerns!

Reply:
Great!